cpp/CMakeLists.txt

```diff
@@ -246,7 +250,20 @@ file(INSTALL ${LIBCXX_INCLUDE_DIR}     DESTINATION ${CMAKE_BINARY_DIR}/include/l
 ###################################################################################################
 # - find zlib -------------------------------------------------------------------------------------
 
-find_package(ZLIB REQUIRED)
+# To retain the variables from system detected packages, I am seperating the logic using find_package
+# then adding if not found.
+# https://github.com/TheLartians/CPM.cmake/issues/132#issuecomment-644955140
+find_package(ZLIB QUIET)
+
+if(NOT ZLIB_FOUND)
+  CPMFindPackage(
+    NAME ZLIB
+    GITHUB_REPOSITORY madler/zlib
+    VERSION 1.2.11
+    )
+
+  set(ZLIB_LIBRARIES ZLIB)
+endif()
```

Comment:
We should be able to just use `CPMFindPackage(ZLIB)` here instead of having a `find_package` followed by a `CPMFindPackage`. If `ZLIB_LIBRARIES` isn't being set properly in the case of a `find_package` we should either create the variable or move to using a ZLIB target instead.